platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowgraphSynchronizer.java

```diff
@@ -382,6 +388,10 @@ private List<EventImpl> createSendList(
             sendList = eventsTheyMayNeed;
         }
 
+        if (maximumEventsPerSync > 0 && sendList.size() > maximumEventsPerSync) {
+            sendList = sendList.subList(0, maximumEventsPerSync);
```

Comment:
I think its a lot better if we don't even add them to the list in the first place

Reply:
Challenge with that approach is that we gather events out of order, and sort them at the end. It's not obvious to me how to just gather the right amount.

Reply:
yeah, lets not complicate it now